src/test/models/AuthPage.ts

```diff
@@ -0,0 +1,185 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+import { Session, UiNode } from "@ory/client"
+import { expect, Locator, Response } from "@playwright/test"
+import { merge } from "lodash"
+import { sessionForbiddenFixture } from "../fixtures"
+import { MockFlow, MockFlowResponse, Traits } from "../types"
+import { inputNodesToRecord, isUiNode, RandomEmail, UUIDv4 } from "../utils"
+
+export class AuthPage {
+  readonly locator: Locator
+  readonly traits: Record<string, Traits>
+  readonly formFields: Record<string, Locator> = {}
+  readonly flowMessage: Locator
+
+  constructor(traits: Record<string, Traits> | UiNode[], locator: Locator) {
+    this.locator = locator
+    this.traits = isUiNode(traits) ? inputNodesToRecord(traits) : traits
+
+    for (const key in this.traits) {
+      this.formFields[key] = locator.locator(
+        `input[name="${this.traits[key].name || key}"]`,
+      )
+    }
+
+    this.flowMessage = locator.locator("*[data-testid*='ui/message/']")
```

Comment:
```suggestion
    this.flowMessage = locator.getByTestId("ui/message/")
```

Should just work

Reply:
I guarantee you it doesn't :)

It's not the same since the selector above does a wildcard select, doesn't care about the error code, just that the format matches `ui/message/<something>`

![image](https://user-images.githubusercontent.com/18033717/228527962-4dc991b8-8687-45e6-9678-99a69d48f93a.png)

Reply:
Ah, the `data-testid` contains an id? So `data-testid-XX`? Not exactly how that's supposed to be used. But fine with me.

Reply:
Nvm, the attribute is `data-testid="ui/message/123"`. It would certainly be better to test for the id here, and not just some string, but so be it, I guess.

The syntax is still incredibly non-obvious. Why does adding a `*` to `data-testid` make it match `data-testid="ui/message/123"`. Maybe you can add a comment here, because sooner or later someone will get confused on that.